### PR TITLE
Add rules for QEMU/libvirt guests (bridged virtual networking)

### PR DESCRIPTION
**Pull request by nbarrientos**

### CODE REVIEW

**Review by traylenator — DISMISSED** [review]
Good stuff.

File: manifests/rules/qemu.pp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  Boolean                                 $dns               = true,
+  Boolean                                 $dhcpv4            = true,
+  Boolean                                 $forward_traffic   = true,
+  Boolean                                 $internal_traffic  = true,
+  Boolean                                 $masquerade        = true,
+) {
+  if $dns {
+    nftables::rule {
+      'default_in-qemu_udp_dns':
+        content => "iifname \"${interface}\" udp dport 53 accept";
```
Comment (bastelfreak) [inline manifests/rules/qemu.pp:48]: are single quotes allowed as well?
```suggestion
        content => "iifname '${interface}' udp dport 53 accept";
```

I'm not sure if that would make it more readable?

Comment (bastelfreak) [reply manifests/rules/qemu.pp:48]: mhm, does it need to be quoted at all?

Comment (nbarrientos) [reply manifests/rules/qemu.pp:48]: The manpage only mentions double quotes:

> Identifiers begin with an alphabetic character (a-z,A-Z), followed zero or more alphanumeric characters (a-z,A-Z,0-9) and the characters slash (/), backslash (\), underscore (_) and dot (.). Identifiers using different characters or clashing with a keyword need to be enclosed in double quotes (").  

```
@some.where.cern.ch /ssh:some.where.cern.ch: λ sbin/nft add rule inet filter INPUT iifname iifname accept
Error: syntax error, unexpected iifname
add rule inet filter INPUT iifname iifname accept
                                   ^^^^^^^
@some.where.cern.ch /ssh:some.where.cern.ch: λ sbin/nft add rule inet filter INPUT iifname \'iifname\' accept
Error: syntax error, unexpected junk
add rule inet filter INPUT iifname 'iifname' accept
                                   ^
@some.where.cern.ch /ssh:some.where.cern.ch: λ sbin/nft add rule inet filter INPUT iifname \"iifname\" accept
@some.where.cern.ch /ssh:some.where.cern.ch: λ sbin/nft --version
nftables v0.9.3 (Topsy)
```

Comment (nbarrientos) [reply manifests/rules/qemu.pp:48]: > mhm, does it need to be quoted at all?

It does to allow interfaces named after reserved words.

Comment (bastelfreak) [reply manifests/rules/qemu.pp:48]: meh :(

**Review by bastelfreak — DISMISSED** [review]

**Review by duritong — APPROVED** [review]
needs rebase otherwise lgtm

**Comment by nbarrientos** [comment]
Needs approval again after rebasing, one should do.

**Review by duritong — APPROVED** [review]

**Comment by nbarrientos** [comment]
Thanks, @duritong

**Pull request merged**